plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/edit/SQLServerTableManager.java

```diff
@@ -118,7 +120,15 @@ public void renameObject(@NotNull DBECommandContext commandContext, @NotNull SQL
     }
 
     @Override
-    protected boolean isIncludeIndexInDDL(DBRProgressMonitor monitor, DBSTableIndex index) throws DBException {
+    protected boolean isIncludeIndexInDDL(@NotNull DBRProgressMonitor monitor, @NotNull DBSTableIndex index) throws DBException {
+        Collection<? extends DBSEntityConstraint> constraints = index.getTable().getConstraints(monitor);
+        if (constraints.size() > 0 && index.isUnique() && constraints.stream().anyMatch(
+            c -> c instanceof SQLServerTableUniqueKey
+                && c.getConstraintType() == DBSEntityConstraintType.UNIQUE_KEY 
+                && ((SQLServerTableUniqueKey) c).getIndex() == index
+        )) {
+            return false;
+        }
```

Comment:
It's hard to read. Consider rewriting using for-each loop.

Reply:
Ok